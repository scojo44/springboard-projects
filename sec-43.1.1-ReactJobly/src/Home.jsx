import React, {useContext} from 'react'
import UserContext from './UserContext'
import './Home.css'

export default function Home() {
  const {currentUser} = useContext(UserContext);

  return (
    <section className="Home">
      <h2>Welcome, {currentUser? currentUser.firstName : 'guest'}!</h2>
    </section>
  );
}
