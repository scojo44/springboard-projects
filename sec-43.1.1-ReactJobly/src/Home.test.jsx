import {render} from '@testing-library/react'
import {MemoryRouter} from 'react-router-dom'
import UserContext from './UserContext'
import Home from './Home'

const context = {
  currentUser: {firstName: 'Testuser'}
};

describe('Home Tests', () => {
  it('Renders without crashing', () => {
    render(
      <MemoryRouter>
        <UserContext.Provider value={context}>
          <Home />
        </UserContext.Provider>
      </MemoryRouter>
    );
  });

  it('Matches snapshot', () => {
    const {asFragment} = render(
      <MemoryRouter>
        <UserContext.Provider value={context}>
          <Home />
        </UserContext.Provider>
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
