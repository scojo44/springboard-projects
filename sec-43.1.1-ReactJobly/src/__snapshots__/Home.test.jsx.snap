// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home Tests > Matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="Home"
  >
    <h2>
      Welcome, Testuser!
    </h2>
  </section>
</DocumentFragment>
`;
